COMPOUND_COMPONENTS.md

```diff
@@ -0,0 +1,336 @@
+# Compound Components
+
+## What is a compound component?
+
+It is a way of describing a complex component by simultaneously describing sub-components. A compound component is made up of a few different parts.
+
+* Container component
+* Sub-components
+* Shared model (optional)
+* Behavior hooks (optional)
+
+A compound component contrasts with a configuration component.
```

Comment:
I think we need a better definition here and to describe a configuration component.

```suggestion
Compound components is a pattern in which components are used together in such a way that describes a higher level of abstraction. A desktop computer is composed of many components that can be serviced and configured individually. Similarly, a compound component allows simultaneous configuration through access to each sub-component. Here are the parts of a compound component:

* Container component
* Sub-components
* Shared model (optional)
* Behavior hooks (optional)

A compound component contrasts with a configuration component which instead configures from a single interface. A configuration component might be like choosing a desktop computer based on stats - how much RAM or how fast the CPU should be or based on what you want to use it for. A compound component is more like buying the parts individually and assembling yourself.
```